omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,297 @@
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize));
+                    Set<PooledIdentifier> pooledIdentifiers = new LinkedHashSet<PooledIdentifier>();
+                    for(Integer counter = 0; counter < Integer.parseInt(minPoolSize); counter++) {
+                            pooledIdentifiers.add(new PooledIdentifier(identifierSource, "ABC0" + counter));
+                    }
+                    identifierSource.setIdentifiers(pooledIdentifiers);
+                }else {
+                    errors.add("min pool size");
+                }
+            }
+            if (!StringUtils.isBlank(sequential)) {
+                if(sequential.toLowerCase().equals("true") || sequential.toLowerCase().equals("false")) {
+                    identifierSource.setSequential(Boolean.parseBoolean(sequential));
+                }else {
+                    errors.add("sequential");
+                }
+            }
+            if (!StringUtils.isBlank(refillWithScheduledTask)) {
+                if(refillWithScheduledTask.toLowerCase().equals("true") || refillWithScheduledTask.toLowerCase().equals("false")) {
+                    identifierSource.setSequential(Boolean.parseBoolean(refillWithScheduledTask));
+                }else {
+                    errors.add("refill");
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ConversionException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name);
+            identifierSource.setDescription(description);
+            identifierSource.setSource(poolSource);
+
+            savedIdentifierSource =  save(identifierSource);
+
+        }
+        return ConversionUtil.convertToRepresentation(savedIdentifierSource, Representation.DEFAULT);
+    }
+	
+    @Override
+    public IdentifierSource save(IdentifierSource identifierSource) {
+        return Context.getService(IdentifierSourceService.class).saveIdentifierSource(identifierSource);
+    }
+
+    @Override
+    public IdentifierSource getByUniqueId(String uniqueId) {
+        return Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uniqueId);
+    }
+	
+    @Override
+    protected void delete(IdentifierSource identifierSource, String reason, RequestContext context) throws ResponseException {
+        Context.getService(IdentifierSourceService.class).retireIdentifierSource(identifierSource, reason);
+    }
+	
+    @Override
+    public void purge(IdentifierSource identifierSource, RequestContext context) throws ResponseException {
+        Context.getService(IdentifierSourceService.class).purgeIdentifierSource(identifierSource);
+    }
+
+    private String validateInput(Object input) {
+        if (input == null || input.toString().length() == 0) {
```

Comment:
What exactly is validateInput doing? Can you put it in words?

Reply:
It allows me to set a default value for variables not sent by requester as an empty string "" which I can then check for blank; and also return the toString() value of variables which are sent by the requester. Otherwise if I called `toString()` on the `postBody.get()` it would be `null.toString()` which would throw an error.

Reply:
Why would you call null.toString() if you are already doing if (StringUtils.isNotBlank())?

Reply:
If I try to make a call without all parameters (required and not required) and remove validateInput() , the call test fails.

Reply:
Does it still fail when you first do: if (StringUtils.isNotBlank())?

Reply:
Yes. i also need the variables which have not been sent to have a default value of "" so that their setters are not called.

Reply:
Validation usually checks for invalid input and throws an exception. This method seems to be doing something else. Can you convince me why we need it in the presence of utitlities like? StringUtils.isBlank() and StringUtils.isNotBlank()

Reply:
Do you have any response to this?

Reply:
Sorry forgot about this. Fixed.

Reply:
I use this method to check if an input has been sent or not. If I do: `StringUtils.isNotBlank(postBody.get("name"))` and the user does not send the name, the `postBody.get("name")` will return a null value. `StringUtils.isNotBlank(postBody.get("name"))` will still pass because null is not blank. The method sets any inputs not sent by the user to blank `""` then I use `StringUtils.isNotBlank(postBody.get("name"))` later to validate the input.

Reply:
I think its also cleaner than using the ternary operator:

`String name = StringUtils.isEmpty(postBody.get('name'))  ? null : postBody.get('name').toString();`

Reply:
A validate method is not supposed to modify input. It only checks if valid and either return false, list of errors, or throw an exception.
What is the value of this method if the caller can simply do something like below:
String name = postBody.get("name");
If (StringUtils.isNotBlank(name)) { do whatever you want with name} else { anything else}

Reply:
fixed